File: Lecture11/multicast/MulticastSender.java
import java.net.*;
import java.io.*;

public class MulticastSender {

    public static void main(String[] args) throws Exception {

        InetAddress group = InetAddress.getByName("225.0.0.37");
        int port = 10000;
        MulticastSocket socket = new MulticastSocket();
        //socket.joinGroup(group);

        String msg = "This is a multicast message.\n";
        byte[] buf = msg.getBytes();

        System.out.println("The multicast sender is on.");

        while (true) {
            System.out.println("The multicast sender sends: " + msg);
            DatagramPacket packet = new DatagramPacket(buf, buf.length, group, port);
            socket.send(packet);
            Thread.sleep(5000);
        }

    }
}

File: Lecture11/multicast/multicast_sender.c
#include <sys/socket.h>
#include <arpa/inet.h>
#include <strings.h>
#include <stdio.h>
#include <unistd.h>

// This program sends multicast message periodically
int main() {
    int sockfd = socket(PF_INET, SOCK_DGRAM, 0);  

    char msg[] = "This is a multicast message.\n";
    printf("The multicast sender is on.\n");
    struct sockaddr_in dest;
    bzero(&dest, sizeof(dest));
    dest.sin_family = PF_INET;
    //Use the multicast address 225.0.0.37 to multicast a message
    inet_pton(PF_INET,"225.0.0.37",&(dest.sin_addr.s_addr)); 
    dest.sin_port = htons(10000);

    while(1) {
        printf("The multicast sender sends: This is a multicast message.\n");  
        sendto(sockfd, msg, sizeof(msg), 0, (struct sockaddr*) &dest, sizeof(dest));
        sleep(5);
    }
}
